pkg/graphql/request.go

```diff
@@ -156,17 +156,57 @@ func (r *Request) IsIntrospectionQuery() (result bool, err error) {
 		return
 	}
 
-	selectionSet := r.document.SelectionSets[operationDef.SelectionSet]
+	selectionSet = &r.document.SelectionSets[operationDef.SelectionSet]
 	if len(selectionSet.SelectionRefs) == 0 {
 		return
 	}
 
+	return selectionSet, nil
+}
+
+func (r *Request) scanFragmentDefinitionsFindSelectionSets() ([]*ast.SelectionSet, error) {
+	report := r.parseQueryOnce()
+	if report.HasErrors() {
+		return nil, report
+	}
+
+	// See the following constants:
+	//
+	// * inlineFragmentedIntrospectionQueryWithFragmentOnQuery
+	// * inlineFragmentedIntrospectionQueryType
+	// * fragmentedIntrospectionQuery
+
+	var selectionSets []*ast.SelectionSet
+	for i := 0; i < len(r.document.FragmentDefinitions); i++ {
+		fragment := r.document.FragmentDefinitions[i]
+		selectionSet := r.document.SelectionSets[fragment.SelectionSet]
```

Comment:
I am still slightly new to the library, but might want to check if the fragment has a selectionset with `fragment.HasSelection`

Reply:
It's unlikely but if i remember correctly the ref for the selectionset could still be -1 and that would panic this.